### ISSUE TITLE
sub_borrow doesn't have default carry flag handler

### ISSUE THREAD

**Issue by samrussell** [Vector35/binaryninja-api#5123]
**Version and Platform (required):**
- Binary Ninja Version: 3.5.4526
- OS: win10
- OS 10
- CPU x64

**Bug Description:**
Extending the gameboy plugin to implement LLIL lifting, the carry flag appears to process correctly on most arithmetic instructions but sub_borrow raises an error

replicating
0000397c  9a       sbc     A, D

![image](https://github.com/Vector35/binaryninja-api/assets/1481279/438f2a97-1859-405b-bc75-8acffe39c9c1)

in comparison, SUB works fine

00003977  92       sub     D

my guess is this was just an oversight and the flag calculation on LowLevelILFunction.sub() can be copied across (unsigned comparison of rhs and result and if result is larger then we have overflowed and set C)

**Comment by plafosse** [comment]
Does your PR not add support this?

**Comment by samrussell** [comment]
Nope, the my PR for the Z80 plugin is a workaround, but this is a binary ninja issue. LLIL_ADC has the carry flag automatically calculated but LLIL_SBB doesn't. From what I can see SBB is just the complement to ADC and the fact that ADC is implemented means SBB should also be implemented in the default `Architecture` implementation
